util/timer/src/timer.rs

```diff
@@ -320,6 +378,14 @@ impl SchedulerInner {
         }
     }
 
+    fn clear_and_stop(&mut self) {
+        self.finished = true;
+        self.states.clear();
```

Comment:
Would you explain why do we need to clear the states?

Reply:
@foriequal0 ?

Reply:
`self.states` is for indexing items in `self.heap`. It is natural to `drain/clear` them in pair.